### PR TITLE
Better image viewer

### LINKED ISSUE #9896: Image Viewer: Zed should jump to an existing tab instead of creating a duplicate
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When clicking on an image file in the explorer, Zed should jump to the image tab, if the image is already open, instead of creating a new copy of the tab. This can be quickly reproduced by clicking several times on an image file in the project panel.

### Environment

Zed: v0.128.3 (Zed)
OS: macOS 14.4.1
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

### LINKED ISSUE #11529: Image preview does not detect file changes and update
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The PNG image preview inside Zed does not update when the image changes.

### Environment

Zed: v0.133.7 (Zed)
OS: macOS 14.4.1
Memory: 24 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by bennetbo**
This fixes some issues with the image viewer:

- Closes: https://github.com/zed-industries/zed/issues/9896
- Closes: https://github.com/zed-industries/zed/issues/11529

TODO:
- Rework file watching to use the built-in UpdateEntriesSet

Release Notes:

- N/A *or* Added/Fixed/Improved ...

### CODE REVIEW

**Comment by bennetbo** [comment]
#19717 resolves the first issue.
Implementing file watching for images requires us to build a `BufferStore` equivalent for images, since that's a larger undertaking, I'm going to close this for now and start from scratch when we find the time.

File: crates/image_viewer/src/image_viewer.rs [hunk 1]
```diff
@@ -45,41 +55,125 @@ impl project::Item for ImageItem {
...
+            .detach();
+            ImageItem {
+                id,
+                abs_path,
+                project_path,
+                image,
+            }
+        })
+    })
+}
```
Comment (rgbkrk) [inline crates/image_viewer/src/image_viewer.rs:124]: Rad. Thanks.

File: crates/image_viewer/src/image_viewer.rs [hunk 2]
```diff
@@ -45,41 +55,125 @@ impl project::Item for ImageItem {
...
+        id: hash(&new_contents),
+        format: match format {
+            image::ImageFormat::Png => gpui::ImageFormat::Png,
+            image::ImageFormat::Jpeg => gpui::ImageFormat::Jpeg,
+            image::ImageFormat::Gif => gpui::ImageFormat::Gif,
+            image::ImageFormat::WebP => gpui::ImageFormat::Webp,
+            image::ImageFormat::Tiff => gpui::ImageFormat::Tiff,
+            image::ImageFormat::Bmp => gpui::ImageFormat::Bmp,
+            _ => {
+                log::error!("Image format not supported");
```
Comment (rgbkrk) [inline crates/image_viewer/src/image_viewer.rs:144]: Should we log the `abs_path` if the image format isn't supported?

**Comment by rgbkrk** [comment]
Oops, I just noticed you closed this. I left this tab open with some starting review comments that I just went ahead and posted. Looking forward to the next PR!

**Pull request closed without merging**